### PR TITLE
Update fields to use mapstr in vSphere virtualmachine metricset

### PR DESCRIPTION
**Pull request by ishleenk17**
## Proposed commit message

See title

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -61,6 +61,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Mark system process metricsets as running if metrics are partially available {pull}40565[40565]
 - Added back `elasticsearch.node.stats.jvm.mem.pools.*` to the `node_stats` metricset {pull}40571[40571]
 - Add support for snapshot in vSphere virtualmachine metricset {pull}40683[40683]
+- Update fields to use mapstr in vSphere virtualmachine metricset  {pull}40707[40707]
```
Comment (niraj-elastic) [inline CHANGELOG.next.asciidoc:64]: I don't think this is breaking change, since the old PR is not released for users yet.

**Review by niraj-elastic — CHANGES_REQUESTED** [review]

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (shmsr) [reply CHANGELOG.next.asciidoc:64]: yeah.

File: metricbeat/module/vsphere/virtualmachine/_meta/data.json [hunk 2]
```diff
@@ -14,73 +14,78 @@
...
+            "Description": "Test snapshot 1",
+            "CreateTime":  "2024-09-01T12:34:56Z",
+            "State":       "PoweredOff"
+            },
+            {
+            "ID":          745,
+            "Name":        "Snapshot_2",
+            "Description": "Test snapshot 2",
+            "CreateTime":  "2024-09-03T2:34:56Z",
+            "State":       "PoweredOn"
```
Comment (shmsr) [inline metricbeat/module/vsphere/virtualmachine/_meta/data.json:84]: shouldn't these fields be in smallcase?

File: metricbeat/module/vsphere/virtualmachine/virtualmachine.go [hunk 3]
```diff
@@ -61,11 +61,11 @@ type VMData struct {
...
 type VMSnapshotData struct {
-	ID          int32
-	Name        string
-	Description string
-	CreateTime  time.Time
-	State       types.VirtualMachinePowerState
+	ID          int32                          `json:"id"`
+	Name        string                         `json:"name"`
+	Description string                         `json:"description"`
+	CreateTime  time.Time                      `json:"createtime"`
```
Comment (shmsr) [inline metricbeat/module/vsphere/virtualmachine/virtualmachine.go:67]: createtime/ create_time?

File: CHANGELOG.next.asciidoc (hunk 1 above)
Comment (ishleenk17) [reply CHANGELOG.next.asciidoc:64]: My bad. This hsould not be here.
Also, I noticed our other PR changelogs are not in properly placed as well.
Some are in filebeat. Some under breaking change. 
We need to recheck this @kush-elastic @niraj-elastic

File: metricbeat/module/vsphere/virtualmachine/virtualmachine.go (hunk 3 above)
Comment (ishleenk17) [reply metricbeat/module/vsphere/virtualmachine/virtualmachine.go:67]: createtime should be ok.

File: metricbeat/module/vsphere/virtualmachine/_meta/data.json (hunk 2 above)
Comment (ishleenk17) [reply metricbeat/module/vsphere/virtualmachine/_meta/data.json:84]: yes, I'll change the json

File: metricbeat/module/vsphere/virtualmachine/_meta/data.json [hunk 4]
```diff
@@ -14,73 +14,78 @@
...
-                "mhz": 0
-              },
-              "total": {
-                "mhz": 0
-              }
+      "virtualmachine": {
+        "name": "xt0nmfpv9",
+        "uptime": 5348978,
+        "host.id": "host-32",
+        "host.hostname": "phx-w1c1-esxi04.com",
```
[the remark this answers is not in the document]
Comment (ishleenk17) [reply metricbeat/module/vsphere/virtualmachine/_meta/data.json:21]: The host to VM association is 1:1. So, it wouldn't be an array.
I can group the id and the hostname

**Review by niraj-elastic — APPROVED** [review]
LGTM

**Pull request merged**